src/common/checksum.c

```diff
@@ -1156,6 +1155,7 @@ ci_insert(struct dcs_csum_info *dcb, int idx, uint8_t *csum_buf, size_t len)
 {
 	uint8_t *to_update;
 
+	D_ASSERT(dcb != NULL);
```

Comment:
do we know if this assertion can be true? can this be triggered by any user input (corrupt or not) or in other circumstances?
is there no recovery beyond this point to assert the engine?

Reply:
I don't think it can, but you make a good point. I'll change it to log an error.